src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -4055,9 +4055,10 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
         NamedIntrinsic nestedOp2Id = nestedOp2->AsHWIntrinsic()->GetHWIntrinsicId();
 
         // If the nested op uses Pg/Z, then inactive lanes will result in zeros, so can only transform if
-        // op3 is all zeros.
+        // op3 is all zeros. Such a Csel operation is absorbed into the instruction when emitted. Skip this optimisation
+        // when the nestedOp is a reduce operation.
 
-        if (nestedOp1->IsMaskAllBitsSet() &&
+        if (nestedOp1->IsMaskAllBitsSet() && !HWIntrinsicInfo::IsReduceOperation(nestedOp2Id) &&
```

Comment:
The reduction operations are notably a case where the pattern we want is somewhat inversed.

Rather than `CndSel(mask, Op(value), merge)` we want to instead want to check for `Op(CndSel(mask, value, zero))` or similar.

We should have a tracking issue to ensure that happens for .NET 10

Reply:
Being looked at in #101973